pinot-core/src/main/java/org/apache/pinot/server/realtime/ControllerLeaderLocator.java

```diff
@@ -41,20 +43,25 @@
   private static ControllerLeaderLocator _instance = null;
   public static final Logger LOGGER = LoggerFactory.getLogger(ControllerLeaderLocator.class);
 
+  // Minimum millis which must elapse between consecutive invalidation of cache
+  private static final long MILLIS_BETWEEN_INVALIDATE = 30_000L;
+
   private final HelixManager _helixManager;
 
-  // Co-ordinates of the last known controller leader.
-  private Pair<String, Integer> _controllerLeaderHostPort = null;
+  // Co-ordinates of the last known controller leader for each of the lead-controller every partitions,
+  // with partition number being the key and controller hostname and port pair being the value.  If the lead
+  // controller resource is disabled in the configuration then this map contains helix cluster leader co-ordinates
+  // for all partitions of leadControllerResource.
+  private final Map<Integer, Pair<String, Integer>> _cachedControllerLeaderMap;
 
-  // Indicates whether cached controller leader value is invalid
-  private volatile boolean _cachedControllerLeaderInvalid = true;
+  // Indicates whether cached controller leader(s) value is(are) invalid.
```

Comment:
lol let's revert this comment change, and reversed boolean logic?

Reply:
Done lol